Support pretty print Fluent::Config::Section for debugging

kenhys opened the pull request:
**Which issue(s) this PR fixes**: 

Fixes N/A

**What this PR does / why we need it**: 

If pp instance of Fluent::Config::Section, it raises the following
exception:

  Unexpected error undefined method `pretty_print' for
  <Fluent::Config::Section {}>

This commit fixes such a case to make debugging easy.

(When debugging https://github.com/fluent/fluentd/pull/3396, I've found
that this feature is missing)

**Docs Changes**:

N/A

**Release Note**: 

N/A

ashie reviewed (commented):
> If pp instance of Fluent::Config::Section,

What's this commit message mean?

kenhys commented:
> > If pp instance of Fluent::Config::Section,
> 
> What's this commit message mean?

Oops, it is an incomplete , broken sentence. I've rebased and modified commit message again.

ashie reviewed (approved):
LGTM
